src/DAL.Infrastructure/Cache/CacheCleaner.cs

```diff
@@ -0,0 +1,32 @@
+﻿using DAL.Events;
+using DAL.Infrastructure.Cache;
+using StackExchange.Redis;
+
+namespace OrderApplication.Cache
+{
+	internal class CacheCleaner : ICacheCleaner
+	{
+		private readonly IDatabase _database;
+
+		public CacheCleaner(IDatabase database)
+		{
+			_database = database;
+		}
+
+		public async Task CleanEventSeatsCache(IList<EventSeat> seats)
+		{
+			if (seats.Any(s => s?.Seat?.Row is null))
+				return;
+
+			var keys = new HashSet<string>();
+			foreach (var seat in seats)
+			{
+				keys.Add(string.Format(EventCacheKeysTemplates.EventAppEventSeatsByEventIdSectionIdCacheTemplate,
+					seat.EventId,
+					seat.Seat.Row.SectionId));
+			}
+
+			await _database.KeyDeleteAsync(keys.Select(k => new RedisKey(k)).ToArray());
+		}
```

Comment:
with deleting cache keys in bulk like this you asume, that all those keys should be stored in a single node, but for this purpose your keys should be properly tagged, otherwise it will not work on real environment. Also storing keys in a single node can lead to this node overload and decreasing performance on get operations.

Reply:
I have little understanding of such problems. I think that's a topic I would like to learn about.